Pipeline/SimplePackageDeploy/SimplePackageDeploy.groovy

```diff
@@ -0,0 +1,276 @@
+ * --hlq <hlq>                 HLQ of the target environment libraries for the deployment 
+ * --packageWithExtension      Flag to show the package contains extension (Optional) 
+ * 
+ * SimplePackageDeploy.properties:
+ * copyModeMap                 JSON string of deploy type and copy mode pairs
+ * targetLibLLQMap             JSON string of deploy type and target LLQ pairs 
+ *
+ * INTERNALS: 
+ *
+ *  1) Reads the options --workDir, --tarFileName, and --hlq from the command line. Pass the optional 
+ *     parm --packageWithExtension if the build package is created with addExtension option. 
+ *  2) Reads the options copyModeMap and targetLibLLQMap from the SimplePackageDeploy.properties 
+ *     file.
+ *  3) Creates a temporary package extract folder in the path provided using the 
+ *     --workDir option.
+ *  4) Untars the deployment package specified by --tarFileName to the temporary 
+ *     folder created in the previous step.
+ *  5) Reads the BuildReport.json file from the temporary folder.
+ *  6) Identifies all the components to be deployed from the BuildReport.json file.
+ *  7) Identifies deploy types for the components to be deployed from the 
+ *     BuildReport.json file.
+ *  8) Identifies the target libraries based on the deploy type by concatenating the
+ *     --hlq value with the correspoding LLQ value from the JSON string - targetLibLLQMap.
+ *  9) Identifies the copy mode based on the deploy type and corresponding copy mode 
+ *     from the JSON string - copyModeMap.
+ * 10) Copies the components using the copy mode to the respective target libraries.
+ * 11) Deletes the temporary package extract folder after deployment.
+ *
+ * RETURNS:
+ *    Exit Status 0       -  Success
+ *    Exit Status 1       -  Failure
+ *
+ ************************************************************************************/
+
+@Field Properties props = null
+props = parseInput(args)
+
+def startTime = new Date()
+props.startTime = startTime.format("yyyyMMdd.hhmmss.mmm")
+println("** SimplePackageDeploy start at $props.startTime")
+
+// Map of last level dataset qualifier to DBB CopyToPDS CopyMode.
+def copyModeMap = parseJSONStringToMap(props.copyModeMap)
+// Map of last level dataset qualifier to Target PDS last level qualifier.
+def targetLibLLQMap = parseJSONStringToMap(props.targetLibLLQMap)
+
+// Define and create tar file extract directory 
+def tarExtractDirName = "${props.workDir}/DeployFiles_${props.startTime}"
+def tarExtractDir = new File("${tarExtractDirName}")
+
+!tarExtractDir.exists() ?: tarExtractDir.deleteDir()
+tarExtractDir.mkdirs()
+println("** Created tar file extract directory $tarExtractDirName")       
+
+// Check if the tar file provided is existing 
+def tarFile = new File("${props.tarFileName}")
+
+if(!tarFile.exists()){
+    println("** Tar file not found at $tarFile")
+    System.exit(1)
+}
+
+
+// Untar the tar file
+println("** Untar file at $tarFile.")
```

Comment:
I was running your script, and this output in the log confused me.

```
** Untar file at /u/dbehm/test-zapp-app-out/mortgageout/build.20221111.022101.021/build.20221111.022101.021.tar.
Package untar done to /u/dbehm/git/anup/work/DeployFiles_20221111.025913.059
```

I would rather rephrase this to
```
** Untarring file /u/dbehm/test-zapp-app-out/mortgageout/build.20221111.022101.021/build.20221111.022101.021.tar. to /u/dbehm/git/anup/work/DeployFiles_20221111.025913.059
```

Reply:
Done